sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -174,6 +228,12 @@ struct _pi_context : _pi_object {
   // and destroy the pool if there are no alive events.
   ze_result_t decrementAliveEventsInPool(ze_event_pool_handle_t pool);
 
+  // Store USM allocator context(internal allocator structures)
+  // for USM shared/host and device allocations. There is 1 allocator context
+  // per each pair of (context, device) per each memory type.
```

Comment:
Why there should be separate allocators for device and shared USM?

Reply:
They have different allocation function: zeMemAllocShared vs zeMemAllocDevice. 
Technically, it's possible to have a single allocator for both of them, but it requires to differentiate and keep track of allocation type. It basically makes the core logic more complex without any advantages of the current approach where we keep allocator per allocation type.

Reply:
OK, it make sense.

Reply:
Yes, keeping track of them separately is good.